ocaml/xapi/pool_features_helpers.ml

```diff
@@ -0,0 +1,108 @@
+(*
+ * Copyright (C) 2024 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Features
+
+module D = Debug.Make (struct let name = "pool_features_helpers" end)
+
+open D
+
+(*
+	Terminology:
+	- (Feature) flags: The keys in pool.restriction and host.license_params. Strings like "restrict_dmc".
+	- Params: An instance of host.license_params.
+	- Restrictions: A (string * string) list of feature flag to a Boolean string value ("true" or "false").
+	- Features: Values of type Features.feature.
+	- Core: Relating to features known by xapi, as define in features.ml.
+	- Additional: Relating to features provided by v6d beyond the core ones.
+*)
+
+let all_flags = List.map (fun (k, _) -> k) (to_assoc_list all_features)
```

Comment:
Maybe map `fst`, as done elsewhere in the code.

Reply:
Aha, I just simply relocated the old code to another file, I`ll refrain from making any modification to it